virt/xen/xen.py

```diff
@@ -0,0 +1,294 @@
+    A model for information about a hypervisor
+    """
+    def __init__(self, hypervisorId,  guestIds=None, name=None, facts=None):
+        """
+        Create a new Hypervisor that will be sent to subscription manager
+
+        'hypervisorId': the unique identifier for this hypervisor
+
+        'guestIds': a list of Guests
+
+        'name': the hostname, if available
+        """
+        self.hypervisorId = hypervisorId
+        self.guestIds = guestIds or []
+        self.name = name
+        self.facts = facts
+
+
+    def toDict(self):
+        d = dict((
+            ('hypervisorId', {'hypervisorId': self.hypervisorId}),
+            ('name', self.name),
+            ('guestIds', [g.toDict() for g in self.guestIds])
+        ))
+        if self.name is None:
+            del d['name']
+        if self.facts is not None:
+            d['facts'] = self.facts
+        return d
+
+    def __str__(self):
+        return str(self.toDict())
+
+    def getHash(self):
+        sortedRepresentation = json.dumps(self.toDict(), sort_keys=True)
+        return hashlib.md5(sortedRepresentation).hexdigest()
+
+class Xen(virt.Virt):
+    CONFIG_TYPE = "xen"
+    # if no events occur the call to event_from will return after this interval
+    # if events do occur the call will return sooner
+    EVENT_FROM_TIMEOUT = 30.0
+
+    # if the token parameter is set to an empty string, the return from event_from will contain all
+    # events that have occurred, you probably want to use this the first time you use event_from
+    token_from =''
+
+    # Register for events on all classes
+    event_types = ["*"]
+
+    def __init__(self, logger, config):
+        super(Xen, self).__init__(logger, config)
+        self.url = config.server
+        self.username = config.username
+        self.password = config.password
+        self.config = config
+
+        # Url must contain protocol (usualy https://)
+        if "://" not in self.url:
+            self.url = "https://%s" % self.url
+
+        self.filter = None
+
+    def _prepare(self):
+        """ Prepare for obtaining information from Xen server. """
+        self.logger.debug("Log into XEN pools %s" % self.url)
+        self.login()
+
+    def login(self):
+        # Login to server using given credentials
+        try:
+            # Don't log message containing password
+            self.session = XenAPI.Session(self.url)
+            self.session.xenapi.login_with_password(self.username, self.password)
+            self.logger.debug("I'm in XEN pools! with user %s" % self.username)
+
+        except NewMaster as nm:
+            self.logger.debug("Tengo un nuevo master?")
+            try:
+                self.session = XenAPI.Session('http://%s' % nm.new_master())
+                self.session.xenapi.login_with_password(self.username, self.password)
+                self.logger.debug("I'm in XEN pools with new master %s! with user %s" % (nm.new_master(), self.username))
+
+            except:
```

Comment:
Do not use plain 'except', you should specify what exceptions you expect here or at lease use 'except Exception:'.